specs/fault-proof/stage-one/optimism-portal.md

```diff
@@ -0,0 +1,251 @@
+would be reported by a perfect oracle for the query.
+
+### Invalid Withdrawal
+
+An **Invalid Withdrawal** is any withdrawal that is not a [Valid Withdrawal](#valid-withdrawal).
+
+### Respected Game Type
+
+See [Respected Game Type](./anchor-state-registry.md#respected-game-type). The Respected Game Type
+can only be set by the Guardian.
+
+### Retirement Timestamp
+
+Any game whose creation timestamp is less than or equal to the **Retirement Timestamp** is
+considered to be a [Retired Game](./anchor-state-registry.md#retired-game). Retired Games cannot be
+used to [prove](#proven-withdrawal) or [finalize](#finalized-withdrawal) withdrawals. The
+Retirement Timestamp can only be set by the Guardian.
+
+### L2 Withdrawal Sender
+
+The **L2 Withdrawal Sender** is the address of the account that triggered a given withdrawal
+transaction on L2. The `OptimismPortal` is expected to expose a variable that includes this value
+when [finalizing](#finalized-withdrawal) a withdrawal.
+
+## Assumptions
+
+### aOP-001: Dispute Game contracts properly report important properties
+
+We assume that the `FaultDisputeGame` and `PermissionedDisputeGame` contracts properly and
+faithfully report the following properties:
+
+- Game type
+- L2 block number
+- Root claim value
+- Game extra data
+- Creation timestamp
+- Resolution timestamp
+- Resolution result
+- Whether the game was the respected game type at creation
+
+We also specifically assume that the game creation timestamp and the resolution timestamp are not
+set to values in the future.
+
+#### Mitigations
+
+- Existing audit on the `FaultDisputeGame` contract
+- Integration testing
+
+### aOP-002: DisputeGameFactory properly reports its created games
+
+We assume that the `DisputeGameFactory` contract properly and faithfully reports the games it has
+created.
+
+#### Mitigations
+
+- Existing audit on the `DisputeGameFactory` contract
+- Integration testing
+
+### aOP-003: Incorrectly resolving games will be invalidated within the airgap delay period
+
+We assume that any games that are resolved incorrectly will be invalidated within the airgap delay
+period.
+
+#### Mitigations
+
+- Stakeholder incentives / processes
+- Incident response plan
+- Monitoring
+
+## Invariants
+
+### iOP-001: Invalid Withdrawals can never be finalized
+
+We require that [Invalid Withdrawals](#invalid-withdrawal) can never be
+[finalized](#finalized-withdrawal) for any reason.
+
+#### Impact
+
+**Severity: Critical**
+
+If this invariant is broken, any number of arbitrarily bad outcomes could happen. Most obviously,
+we would expect all bridge systems relying on the `OptimismPortal` to be immediately compromised.
+
+### iOP-002: Valid Withdrawals can always be finalized in bounded time
+
+We require that [Valid Withdrawals](#valid-withdrawal) can always be
+[finalized](#finalized-withdrawal) within some reasonable, bounded amount of time.
```

Comment:
This invariant is broken by the ability to pause withdrawals.

Reply:
Yeah, that's correct. We need to clarify the assumption that the Guardian will unpause within a bounded amount of time.